diagram.schema.json

```diff
@@ -0,0 +1,240 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "Diagram",
+  "type": "object",
+  "required": [
+    "ops"
+  ],
+  "properties": {
+    "ops": {
+      "type": "object",
+      "additionalProperties": {
+        "$ref": "#/definitions/DiagramOperation"
+      }
+    }
+  },
+  "definitions": {
+    "DiagramOperation": {
+      "oneOf": [
+        {
+          "description": "Signifies the start of a workflow. There must be exactly 1 start operation in a diagram.",
+          "type": "object",
+          "required": [
+            "next",
+            "type"
+          ],
+          "properties": {
+            "next": {
+              "type": "string"
+            },
+            "type": {
+              "type": "string",
+              "enum": [
+                "start"
+              ]
+            }
+          }
+        },
+        {
+          "description": "Signifies the end of a workflow. There must be exactly 1 terminate operation in a diagram.",
+          "type": "object",
+          "required": [
+            "type"
+          ],
+          "properties": {
+            "type": {
+              "type": "string",
+              "enum": [
+                "terminate"
+              ]
+            }
+          }
+        },
+        {
+          "description": "Connects the request to a registered node.\n\n``` # bevy_impulse::Diagram::from_json_str(r#\" { \"ops\": { \"start\": { \"type\": \"start\", \"next\": \"nodeOp\" }, \"nodeOp\": { \"type\": \"node\", \"nodeId\": \"myNode\", \"next\": \"terminate\" }, \"terminate\": { \"type\": \"terminate\" } } } # \"#)?; # Ok::<_, serde_json::Error>(())",
```

Comment:
I'm curious: Is there an IDE or other tool that renders this description string, and if so what does this string look like when rendered? As raw text I find it difficult to interpret.

Reply:
In vscode it will render the the newlines but it doesn't seem to render the markdown, schemars also doesn't seem to generate the formatting properly. I guess we should remove the examples from the schema to make it play nicer.